log.go

```diff
@@ -0,0 +1,80 @@
+package logruswindows
+
+import (
+	"strconv"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows/svc/eventlog"
+)
+
+// EventHook sends logs to window event logs
+type EventHook struct {
+	*eventlog.Log
+	source string
+	levels []logrus.Level
+}
+
+// NewEventHook creates an event logging hook from even source
+// and supported log levels
+func NewEventHook(source string, levels []logrus.Level) (*EventHook, error) {
+	const supports = eventlog.Error | eventlog.Warning | eventlog.Info
+	if err := eventlog.InstallAsEventCreate(source, supports); err != nil {
+		return nil, errors.Wrapf(err, "eventlog.InstallAsEventCreate source=%s", source)
+	}
+
+	l, err := eventlog.Open(source)
+	if err != nil {
+		return nil, errors.Wrapf(err, "eventlog.Open source=%s", source)
+	}
+	return &EventHook{
+		Log:    l,
+		source: source,
+		levels: levels,
+	}, nil
+}
+
+// Fire extracts logrus entry and sends to window event log
+func (hook *EventHook) Fire(entry *logrus.Entry) error {
+	msg, err := entry.String()
+	var eventID uint32 = 1
+	entryEventID, ok := entry.Data["event_id"]
+	if ok {
+		switch id := entryEventID.(type) {
+		case int:
```

Comment:
they're all ignored by now, other hook such as sentry logrus support string only, I think we should pass event id as string also, for compatibility